Description: Enable linking pgsqlclient with linker set for --as-needed
Author: Felix Geyer <fgeyer@debian.org>
Date: Sun, 29 Jul 2012 22:07:33 +0200
Bugs-Debian: http://bugs.debian.org/650237
LastChanged: Mon, 09 Sep 2013 14:48:22 +0200 (Andreas Tille, adapt to 6.6.0)

--- a/m4/postgresql.m4
+++ b/m4/postgresql.m4
@@ -98,10 +98,10 @@ AC_DEFUN([AX_LIB_POSTGRESQL],
       dnl not be installed
 
       EMBCPPFLAGS="${CPPFLAGS}"
-      EMBLDFLAGS="${LDFLAGS}"
+      EMBLIBS="${LIBS}"
 
       CPPFLAGS="${POSTGRESQL_CPPFLAGS} ${EMBCPPFLAGS}"
-      LDFLAGS="${POSTGRESQL_LDFLAGS} ${EMBLDFLAGS}"
+      LIBS="${POSTGRESQL_LDFLAGS} ${EMBLIBS}"
 
       AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <stdio.h>
                                         #include "libpq-fe.h"]],
@@ -110,7 +110,7 @@ AC_DEFUN([AX_LIB_POSTGRESQL],
         [havepostgresql="no"])
 
       CPPFLAGS="${EMBCPPFLAGS}"
-      LDFLAGS="${EMBLDFLAGS}"
+      LIBS="${EMBLIBS}"
 
       AS_IF([test "x${havepostgresql}" = "xyes"],
       [
